lib/models/event-context.js

```diff
@@ -0,0 +1,104 @@
+"use strict";
+
+/**
+ * @module models/event-context
+ */
+
+/**
+ * Construct a new EventContext
+ *
+ * An eventcontext is used for circumstances such as search results, when we
+ * have a particular event of interest, and a bunch of events before and after
+ * it.
+ *
+ * It also stores pagination tokens for going backwards and forwards in the
+ * timeline.
+ *
+ * @param {MatrixEvent} our_event  the event at the centre of this context
+ *
+ * @constructor
+ */
+function EventContext(our_event) {
+    this._timeline = [our_event];
+    this._ourEventIndex = 0;
+    this._paginateTokens = {};
```

Comment:
Would be a good idea to put a one-liner explaining what the key/values here should be as it is not immediately clear where these tokens appear.